From 589b4a4128106adfa80532249cc86ca60ac367fd Mon Sep 17 00:00:00 2001
From: Charles Steinkuehler <charles@steinkuehler.net>
Date: Sun, 28 Sep 2014 12:16:16 +0000
Subject: [PATCH 21/21] pinmux: use hdmi mode

BeagleBone DTS : Enable run-time pinmux for HDMI Add cape-universal-hdmi pin
info to am335x-bone-common-pinmux.dtsi Edit hdmi dtsi include files to use
new mode= setting to set HDMI mode at startup, leaving  pinmux configurable
at runtime.

Signed-off-by: Charles Steinkuehler <charles@steinkuehler.net>
Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/boot/dts/am335x-bone-common-pinmux.dtsi   |  6 ++
 .../boot/dts/am335x-boneblack-nxp-hdmi-audio.dtsi  | 73 +++++++++++++++++++---
 .../dts/am335x-boneblack-nxp-hdmi-no-audio.dtsi    | 61 +++++++++++++++++-
 3 files changed, 128 insertions(+), 12 deletions(-)

diff --git a/arch/arm/boot/dts/am335x-bone-common-pinmux.dtsi b/arch/arm/boot/dts/am335x-bone-common-pinmux.dtsi
index 0be2602..6cf2de1 100644
--- a/arch/arm/boot/dts/am335x-bone-common-pinmux.dtsi
+++ b/arch/arm/boot/dts/am335x-bone-common-pinmux.dtsi
@@ -406,6 +406,12 @@
 		>;
 	};
 
+	mcasp0_clken: mcasp0_clken {
+		pinctrl-single,pins = <
+			0x06c (PIN_OUTPUT_PULLDOWN | MUX_MODE7)	/* gpmc_a11.GPIO1_27 */
+		>;
+	};
+
 	/* audio rev a */
 	mcasp0_pins_audio_reva: mcasp0_pins_audio_reva {
 		pinctrl-single,pins = <
diff --git a/arch/arm/boot/dts/am335x-boneblack-nxp-hdmi-audio.dtsi b/arch/arm/boot/dts/am335x-boneblack-nxp-hdmi-audio.dtsi
index 6466560..ed9861e 100644
--- a/arch/arm/boot/dts/am335x-boneblack-nxp-hdmi-audio.dtsi
+++ b/arch/arm/boot/dts/am335x-boneblack-nxp-hdmi-audio.dtsi
@@ -8,27 +8,81 @@
 
 &ocp {
 	/* hdmi */
+	P8_27_pinmux {
+		mode = "hdmi";
+	};
+	P8_28_pinmux {
+		mode = "hdmi";
+	};
+	P8_29_pinmux {
+		mode = "hdmi";
+	};
+	P8_30_pinmux {
+		mode = "hdmi";
+	};
+	P8_31_pinmux {
+		mode = "hdmi";
+	};
+	P8_32_pinmux {
+		mode = "hdmi";
+	};
+	P8_33_pinmux {
+		mode = "hdmi";
+	};
+	P8_34_pinmux {
+		mode = "hdmi";
+	};
+	P8_35_pinmux {
+		mode = "hdmi";
+	};
+	P8_36_pinmux {
+		mode = "hdmi";
+	};
 	P8_37_pinmux {
-		status = "disabled";
+		mode = "hdmi";
 	};
 	P8_38_pinmux {
-		status = "disabled";
+		mode = "hdmi";
+	};
+	P8_39_pinmux {
+		mode = "hdmi";
+	};
+	P8_40_pinmux {
+		mode = "hdmi";
+	};
+	P8_41_pinmux {
+		mode = "hdmi";
+	};
+	P8_42_pinmux {
+		mode = "hdmi";
+	};
+	P8_43_pinmux {
+		mode = "hdmi";
+	};
+	P8_44_pinmux {
+		mode = "hdmi";
+	};
+	P8_45_pinmux {
+		mode = "hdmi";
+	};
+	P8_46_pinmux {
+		mode = "hdmi";
 	};
 	/* audio */
 	P9_25_pinmux {
-		status = "disabled";
+		mode = "audio";
 	};
 	P9_28_pinmux {
-		status = "disabled";
+		mode = "audio";
 	};
 	P9_29_pinmux {
-		status = "disabled";
+		mode = "audio";
 	};
 	P9_31_pinmux {
-		status = "disabled";
+		mode = "audio";
 	};
 	P9_41_pinmux {
-		status = "disabled";
+		mode = "audio";
 	};
 };
 
@@ -38,7 +92,7 @@
 
 &mcasp0	{
 	pinctrl-names = "default";
-	pinctrl-0 = <&mcasp0_pins>;
+	pinctrl-0 = <&mcasp0_clken>;
 	status = "okay";
 	op-mode = <0>;	/* MCASP_IIS_MODE */
 	tdm-slots = <2>;
@@ -54,7 +108,8 @@
 		compatible = "ti,tilcdc,slave";
 		i2c = <&i2c0>;
 		pinctrl-names = "default", "off";
-		pinctrl-0 = <&nxp_hdmi_bonelt_pins>;
+		/* pinctrl-0 = <&nxp_hdmi_bonelt_pins>; */
+		pinctrl-0 = <&nxp_hdmi_bonelt_off_pins>;
 		pinctrl-1 = <&nxp_hdmi_bonelt_off_pins>;
 		status = "okay";
 	};
diff --git a/arch/arm/boot/dts/am335x-boneblack-nxp-hdmi-no-audio.dtsi b/arch/arm/boot/dts/am335x-boneblack-nxp-hdmi-no-audio.dtsi
index 63d0214..2085aa0 100644
--- a/arch/arm/boot/dts/am335x-boneblack-nxp-hdmi-no-audio.dtsi
+++ b/arch/arm/boot/dts/am335x-boneblack-nxp-hdmi-no-audio.dtsi
@@ -8,11 +8,65 @@
 
 &ocp {
 	/* hdmi */
+	P8_27_pinmux {
+		mode = "hdmi";
+	};
+	P8_28_pinmux {
+		mode = "hdmi";
+	};
+	P8_29_pinmux {
+		mode = "hdmi";
+	};
+	P8_30_pinmux {
+		mode = "hdmi";
+	};
+	P8_31_pinmux {
+		mode = "hdmi";
+	};
+	P8_32_pinmux {
+		mode = "hdmi";
+	};
+	P8_33_pinmux {
+		mode = "hdmi";
+	};
+	P8_34_pinmux {
+		mode = "hdmi";
+	};
+	P8_35_pinmux {
+		mode = "hdmi";
+	};
+	P8_36_pinmux {
+		mode = "hdmi";
+	};
 	P8_37_pinmux {
-		status = "disabled";
+		mode = "hdmi";
 	};
 	P8_38_pinmux {
-		status = "disabled";
+		mode = "hdmi";
+	};
+	P8_39_pinmux {
+		mode = "hdmi";
+	};
+	P8_40_pinmux {
+		mode = "hdmi";
+	};
+	P8_41_pinmux {
+		mode = "hdmi";
+	};
+	P8_42_pinmux {
+		mode = "hdmi";
+	};
+	P8_43_pinmux {
+		mode = "hdmi";
+	};
+	P8_44_pinmux {
+		mode = "hdmi";
+	};
+	P8_45_pinmux {
+		mode = "hdmi";
+	};
+	P8_46_pinmux {
+		mode = "hdmi";
 	};
 };
 
@@ -25,7 +79,8 @@
 		compatible = "ti,tilcdc,slave";
 		i2c = <&i2c0>;
 		pinctrl-names = "default", "off";
-		pinctrl-0 = <&nxp_hdmi_bonelt_pins>;
+		/* pinctrl-0 = <&nxp_hdmi_bonelt_pins>; */
+		pinctrl-0 = <&nxp_hdmi_bonelt_off_pins>;
 		pinctrl-1 = <&nxp_hdmi_bonelt_off_pins>;
 		status = "okay";
 	};
-- 
2.1.0

